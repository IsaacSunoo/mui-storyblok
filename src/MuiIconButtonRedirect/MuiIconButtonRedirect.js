import React, { createElement, lazy, Suspense } from 'react';
import { withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import Storyblok from '../utils/Storyblok';
import IconButton from '@material-ui/core/IconButton';

const MuiIcon = lazy(() => import('../MuiIcon/MuiIcon'));

/**
 * MuiIconButtonRedirect is used in storyblok redirect to react routes
 */

export const MuiIconButtonRedirect = ({
  history,
  color,
  disableFocusRipple,
  disableRipple,
  edge,
  size,
  rootClass,
  icon,
  redirectRoute,
}) => {
  const components = {
    MuiIcon,
  };

  const styles = Storyblok.arrayToMuiStyles(rootClass);

  const onClick = async () => {
    return history.push(redirectRoute);
  };

  return (
    <IconButton
      onClick={onClick}
      className={styles.root}
      color={color}
      disableFocusRipple={disableFocusRipple}
      disableRipple={disableRipple}
      edge={edge === 'false' ? false : edge}
      size={size}
    >
      <Suspense fallback={<div />}>
        {icon[0] ? createElement(components[icon[0].component], { ...icon[0] }) : null}
      </Suspense>
    </IconButton>
  );
};

export default withRouter(MuiIconButtonRedirect);

MuiIconButtonRedirect.propTypes = {
  /** mui prop: 'default'| 'inherit'| 'primary'| 'secondary' */
  color: PropTypes.string,
  /* mui prop: If true, the keyboard focus ripple will be disabled. disableRipple must also be true. */
  disableFocusRipple: PropTypes.bool,
  /* mui prop: If true, the ripple effect will be disabled. */
  disableRipple: PropTypes.bool,
  /* mui prop: 'start' | 'end' | false */
  edge: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
  /** mui prop: 'small' | "medium" */
  size: PropTypes.string,
  /** stroyblok multiselect of css classes */
  rootClass: PropTypes.arrayOf(PropTypes.string),

  /** redirect route */
  redirectRoute: PropTypes.string.isRequired,

  icon: PropTypes.arrayOf(PropTypes.shape({
    component: PropTypes.string.isRequired,
  })).isRequired,

  /** react history not a storyblok prop */
  history: PropTypes.shape({
    push: PropTypes.func,
  }).isRequired,
};

MuiIconButtonRedirect.defaultProps = {
  color: 'default',
  disableFocusRipple: false,
  disableRipple: false,
  edge: false,
  size: 'medium',
  rootClass: [],
};
