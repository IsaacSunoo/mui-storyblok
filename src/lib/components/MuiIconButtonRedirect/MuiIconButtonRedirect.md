MuiIconButtonRedirect example:

```js noeditor
import MuiIconButtonRedirect from './MuiIconButtonRedirect';
```

```js
const iconButton = [{
  component: 'MuiIconButton',
  icon: [{
    component: 'MuiIcon',
    iconName: 'history',
  }],
}];

<MuiIconButtonRedirect
  redirectRoute=""
  iconButton={iconButton}
/>
```