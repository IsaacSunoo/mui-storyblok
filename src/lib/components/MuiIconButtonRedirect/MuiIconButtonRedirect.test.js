import React from 'react';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { MuiIconButtonRedirect } from './MuiIconButtonRedirect';

function setup(legacyRedirect = true) {
  const props = {

    redirectRoute: 'page-welcome',
    legacyRedirect,
    
    icon: [{
      component: 'MuiIcon',
      iconName: 'android',
    }],

    history: {
      push: jest.fn(),
    },

  };
  const comp = shallow(<MuiIconButtonRedirect {...props} />);

  return { comp, props };
}

describe('<MuiIconButtonRedirect />', () => {
  it('renders MuiIconButtonRedirect', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });

  describe('clicks', () => {
    it('handleClick and calls history push ', async () => {
      const { comp, props } = setup(false);
      comp.find('WithStyles(ForwardRef(IconButton))').at(0).simulate('click');
      expect(props.history.push).toBeCalled();
    });
  });

  test('snapshot', () => {
    const { props } = setup();

    const tree = renderer.create((
      <MemoryRouter>
        <MuiIconButtonRedirect {...props} />
      </MemoryRouter>
    ));
    expect(tree).toMatchSnapshot();
  });
});
