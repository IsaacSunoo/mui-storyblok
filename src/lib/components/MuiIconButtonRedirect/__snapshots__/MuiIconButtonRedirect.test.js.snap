// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MuiIconButtonRedirect /> snapshot 1`] = `
<button
  className="MuiButtonBase-root MuiIconButton-root makeStyles-root-4"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiIconButton-label"
  >
    <span
      aria-hidden={true}
      className="material-icons MuiIcon-root makeStyles-root-17 MuiIcon-colorSecondary"
    >
      android
    </span>
  </span>
</button>
`;
